src/plugins/discover/public/application/view_components/utils/use_index_pattern.ts

```diff
@@ -95,7 +105,8 @@ export const useIndexPattern = (services: DiscoverViewServices) => {
     store,
     toastNotifications,
     uiSettings,
-    query?.dataset,
+    query.dataset,
```

Comment:
query will never be undefined?

Reply:
yes, it will be default query if not passed in